iceoryx2-bb/posix/tests/thread_tests.rs

```diff
@@ -181,19 +181,18 @@ fn thread_cancel_works() {
 
 #[test]
 fn thread_exit_works() {
-    let barrier = Arc::new(Barrier::new(2));
```

Comment:
I already tried the sleep in the main thread and it did not work but I did not try it in the test thread after adding the `!` no return.

It is somehow related to the `thread_exit` call, though. Adding a 1 second sleep to the main thread after the construction of the test thread still leads to an immediate `SIGABRT`. But adding a 1 µs sleep in front of the `thread_exit` makes the test pass.

Adding a barrier is useless in this case since the dtor does a blocking wait anyway. Initially I removed the barrier since I thought there might be a race between `thread_exit` and the barrier on the main thread but it seems to not be the case.

What do you think of adding two additional tests instead of using a barrier here
- the dtor does not block on an thread which does nothing
- the dtor blocks on a thread which sleeps for 100ms